inc/entity.class.php

```diff
@@ -71,6 +71,74 @@ public function post_purgeItem() {
       ]);
    }
 
+   public function calculateGlobalDefault() {
+      if (!isset($_SESSION['credit']['post_update'])) {
+         if ($this->fields['is_default_followup']
+            && $this->fields['is_default_task']
+            && $this->fields['is_default_solution']) {
+               $this->fields['is_default'] = 1;
+         } else if (!$this->fields['is_default_followup']
+            && !$this->fields['is_default_task']
+            && !$this->fields['is_default_solution']) {
+            !$this->fields['is_default'] = 0;
+         } else {
+            $this->fields['is_default'] = 2;
+         }
+         $_SESSION['credit']['post_update'] = true;
+         $this->update($this->fields);
+      } else {
+         unset($_SESSION['credit']['post_update']);
+      }
+   }
+
+   public function post_updateItem($history = 1) {
+      $this->calculateGlobalDefault();
+   }
+
+   public function post_addItem($history = 1) {
+      $this->calculateGlobalDefault();
+   }
```

Comment:
Maybe I am wrong, but doing this computation in prepareInputFor... could be a better solution to not have to use `$_SESSION['credit']['post_update']` to preven infinite update llop.